raiden/network/proxies/token_network.py

```diff
@@ -1182,6 +1183,15 @@ def set_total_withdraw(
                     )
                     raise WithdrawMismatch(msg)
 
+                current_block_number = self.client.get_block("latest").number
```

Comment:
1. Queries should be in the `try` block.
2. You should use `given_block_identifier`

Reply:
Which exceptions shall I catch?

Reply:
None, it's already done. But if `given_block_identifier` is pruned, it'll through a `ValueError`, otherwise, it's valid and we use that block identifier in the `else` block.

Reply:
I at least used `given_block_identifier`.

When I said `try:` but didn't `except` any exceptions, I got a syntax error.

Reply:
Shall I do this?
```
                except Exception:  # pylint: disable=broad-except
                    raise RaidenUnrecoverableError(
                        f"Couldn't get the block for {given_block_identifier}"
                    )
```

Reply:
No no we don't want that. You shouldn't have to add any more excepts. If i understand what you're trying to do correctly, you're introducing a new `try` block... but you should in fact use the existing try block that is a bit above where you placed this line.

Reply:
Ohh, I understand now.  I move the line upwards.